src/d3d12/d3d12_renderer.cpp

```diff
@@ -796,9 +814,62 @@ namespace wr
 							static_cast<D3D12ModelPool*>(model->m_model_pool)->GetIndexStagingBuffer(),
 							0,
 							static_cast<D3D12ModelPool*>(model->m_model_pool)->GetIndexStagingBuffer()->m_size);
+
 						m_bound_model_pool = static_cast<D3D12ModelPool*>(model->m_model_pool);
 						m_bound_model_pool_stride = n_mesh->m_vertex_staging_buffer_stride;
 					}
+
+
+					enum MaterialPBR: unsigned int
+					{
+						PBR_Albedo = 0,
+						PBR_Normal,
+
+						PBR_Count
+					};
+					
+					d3d12::BindDescriptorHeaps(n_cmd_list, { m_rendering_heap }, frame_idx);
+
+					auto material_handle = mesh.second;
+					
+					if (material_handle != m_last_material)
```

Comment:
Abstract all this code to a different function. This render function is becoming really fucking bad.